apps/extension/src/shared/components/numeraires-form.tsx

```diff
@@ -13,39 +12,36 @@ const useNumerairesSelector = (state: AllSlices) => {
     selectedNumeraires: state.numeraires.selectedNumeraires,
     selectNumeraire: state.numeraires.selectNumeraire,
     saveNumeraires: state.numeraires.saveNumeraires,
-    networkChainId: state.network.chainId,
   };
 };
 
 export const NumeraireForm = ({
   isOnboarding,
   onSuccess,
+  chainId,
 }: {
+  chainId?: string;
   isOnboarding?: boolean;
-  onSuccess: () => void | Promise<void>;
+  onSuccess: () => void;
 }) => {
-  const { chainId } = useChainIdQuery();
-  const { selectedNumeraires, selectNumeraire, saveNumeraires, networkChainId } =
-    useStore(useNumerairesSelector);
-
-  // 'chainId' from 'useChainIdQuery' is not available during onboarding,
-  // this forces you to use two sources to guarantee 'chainId' for both settings and onboarding
-  const { numeraires } = useNumeraires(chainId ?? networkChainId);
```

Comment:
Instead of doing both (serving the settings & onboarding simultaneously internally), I changed the component so that chainId is passed in